runtime/appruntime/shared/health/health.go

```diff
@@ -0,0 +1,115 @@
+package health
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"golang.org/x/exp/slices"
+)
+
+// CheckRegistry is a registry of health checks from the API and Infra SDKs
+// and other parts of the runtime.
+type CheckRegistry struct {
+	checks []Check
+	m      sync.Mutex
```

Comment:
Nit: mutex goes before the fields it protects
```suggestion
	m      sync.Mutex
	checks []Check
```